.github/workflows/mac_release.yml

```diff
@@ -153,39 +267,96 @@ jobs:
+      id: setup-python
+      with:
+        python-version: ${{ env.Python_REQUIRED_VERSION }}
+        architecture: ${{ matrix.python-arch }}
+
+    - name: Gather Test Package from Artifacts
+      uses: actions/download-artifact@v4
+      with:
+        name: energyplus-${{ matrix.os }}.tar.gz
+        path: package
+
+    - name: Check Contents
+      shell: bash
+      run: ls
+
+    - name: Check Package contents
+      shell: bash
+      working-directory: package
+      run: ls
+
+    - name: Run Package Tests
+      run: python checkout/scripts/package_tests/runner.py --verbose ${{ matrix.test_key }} package/
+
+    - name: Gather Dmg Package from Artifacts
+      uses: actions/download-artifact@v4
+      with:
+        name: energyplus-${{ matrix.os }}.dmg
+        path: dmg
+
+    - name: Test Dmg Install and Package signing
+      working-directory: ./dmg
+      shell: bash
+      run: |
+        begin_group() { echo -e "::group::\033[93m$1\033[0m"; }
+
+        set -x
+
+        dmg=$(ls EnergyPlus-*.dmg)
+        begin_group "Checking Signature of .dmg"
+        spctl --assess --type open --context context:primary-signature -vvvv $dmg
+        echo "::endgroup::"
+
+        begin_group "Mounting Dmg, and checking signature of installer app"
+        mkdir temp_mount
+        hdiutil attach -mountpoint ./temp_mount/ $dmg
+        filename="${dmg%.*}"
+        spctl --assess --type open --context context:primary-signature -vvvv ./temp_mount/$filename.app
+        echo "::endgroup::"
+
+        begin_group "Installing"
+        sudo ./temp_mount/$filename.app/Contents/MacOS/$filename --accept-licenses --default-answer --confirm-command --root $(pwd)/test_install install
+        hdiutil detach ./temp_mount/
+        echo "::endgroup::"
+
+        begin_group "Quick Check signature of inner executables and binaries"
+        codesign -dvvv ./test_install/energyplus
+        codesign -dvvv ./test_install/libenergyplusapi.dylib
+        codesign -dvvv ./test_install/libpython*.dylib
+        codesign -dvvv ./test_install/ConvertInputFormat
+        codesign -dvvv ./test_install/PostProcess/ReadVarsESO
+        echo "::endgroup::"
+
+        begin_group "Full Check signature of installed DMG for all executables and resolve otool libraries"
+        python ../checkout/scripts/dev/verify_signature.py --otool --otool-out-file otool_info_dmg.json --verbose --install test_install
+        echo "::endgroup::"
+
+        begin_group "Running a simulation with python"
+        ./test_install/energyplus --help
+        ./test_install/energyplus -w ./test_install/WeatherData/USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw -d out ./test_install/ExampleFiles/PythonPluginCustomSchedule.idf
+        echo "::endgroup::"
+
+    - name: Upload otool info as artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: otool_info_dmg_${{ matrix.os }}_${{ matrix.arch }}
+        path: dmg/otool*json
+        if-no-files-found: error
```

Comment:
This is new.

I download the .dmg artifact on the test machine and I:

* Ensure it's properly signed and notarized
* Mount it
* Ensure the installer .app is properly signed and notarized
* Install the ifw installer
* Then I test the resulting folder for codesigning and otool for binaries and executables